tests/parametric/test_headers_baggage.py

```diff
@@ -178,8 +178,19 @@ def test_baggage_remove_all_D011(self, test_library):
             span.remove_all_baggage()
             assert span.get_all_baggage() == {}
 
-    def test_baggage_malformed_headers_D012(self, test_library, test_agent):
+    def _assert_valid_baggage(self, test_library):
+        """
+        Helper function to confirm that a valid baggage header is set
+        when calling dd_make_child_span_and_get_headers.
+        """
+        with test_library:
+            headers = test_library.dd_make_child_span_and_get_headers([["baggage", "foo=valid"]])
+            assert "baggage" in headers.keys()
+
+    def test_baggage_malformed_headers_D012(self, test_library):
         """Ensure that malformed baggage headers are handled properly. Unable to use get_baggage functions because it does not return anything"""
+        Test_Headers_Baggage._assert_valid_baggage(self, test_library)
+
         with test_library:
```

Comment:
Could we use the same "_assert_valid_baggage" for both checks? Adding a new param to the method